### PR TITLE
adding collision object interface to planning_scene interface

### PR DESCRIPTION
**Pull request by sachinchitta**

### CODE REVIEW

File: planning_interface/planning_scene_interface/include/moveit/planning_scene_interface/planning_scene_interface.h [hunk 1]
```diff
@@ -72,6 +74,13 @@ class PlanningSceneInterface
 
   std::map<std::string, geometry_msgs::Pose> getObjectPoses(const std::vector<std::string> &object_ids);  
 
+  /** \brief Add collision objects to the world 
+      Make sure object.operation is set to object.ADD*/
+  void addCollisionObjects(const std::vector<moveit_msgs::CollisionObject> &collision_objects) const;  
+
+  /** \brief Remove collision objects from the world*/
+  void removeCollisionObjects(const std::vector<std::string> object_ids) const;
```
Comment (isucan) [inline planning_interface/planning_scene_interface/include/moveit/planning_scene_interface/planning_scene_interface.h]: This looks good. Just add a & for `object_ids`.

File: planning_interface/planning_scene_interface/src/planning_scene_interface.cpp [hunk 2]
```diff
@@ -190,5 +214,15 @@ class PlanningSceneInterface::PlanningSceneInterfaceImpl
   return impl_->getObjectPoses(object_ids);
 }
 
+void PlanningSceneInterface::addCollisionObjects(const std::vector<moveit_msgs::CollisionObject> &collision_objects) const
+{
+  return impl_->addCollisionObjects(collision_objects);
+}
+
+void PlanningSceneInterface::removeCollisionObjects(const std::vector<std::string> object_ids) const
```
Comment (isucan) [inline planning_interface/planning_scene_interface/src/planning_scene_interface.cpp]: And a & for `object_ids` again.

File: planning_interface/planning_scene_interface/src/planning_scene_interface.cpp [hunk 3]
```diff
@@ -158,10 +159,33 @@ class PlanningSceneInterface::PlanningSceneInterfaceImpl
...
 
+  void addCollisionObjects(const std::vector<moveit_msgs::CollisionObject> &collision_objects) const
+  {
+    moveit_msgs::PlanningScene planning_scene;
+    planning_scene.world.collision_objects = collision_objects;
+    planning_scene.is_diff = true;
+    planning_scene_diff_publisher_.publish(planning_scene);    
+  }
+
+  void removeCollisionObjects(const std::vector<std::string> object_ids) const
```
Comment (isucan) [inline planning_interface/planning_scene_interface/src/planning_scene_interface.cpp]: And a & for `object_ids` again.

Comment (sachinchitta) [inline planning_interface/planning_scene_interface/src/planning_scene_interface.cpp]: Sorry about that. Fixed.

**Pull request merged**